### PR TITLE
keycloak sdk

### PR DESCRIPTION
**Pull request by Matrix2100**

### CODE REVIEW

File: src/Keycloak.Dotnet.Sdk/Dtos/Realms/RealmDto.cs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    public string[]? SupportedLocales { get; set; }
+
+    [JsonPropertyName("defaultLocale")]
+    public string? DefaultLocale { get; set; }
+
+    [JsonPropertyName("identityProviders")]
+    public IdentityProviders[]? IdentityProviders { get; set; }
+
+    [JsonPropertyName("identityProviderMappers")]
+    public IdentityProviderMappers[]? IdentityProviderMappers { get; set; }
```
Comment (l3oferreira) [inline src/Keycloak.Dotnet.Sdk/Dtos/Realms/RealmDto.cs:45]: Record names in singular.

File: src/Keycloak.Dotnet.Sdk/Dtos/Auth/KeycloakAuthConfig.cs [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+        public string? ClientId { get; set; }
+        public string? Username { get; set; }
+        public string? Password { get; set; }
+    }
+
+    public record ServiceAccountAggregation
+    {
+        public string? ClientId { get; set; }
+        public string? ClientSecret { get; set; }
+    }
```
Comment (l3oferreira) [inline src/Keycloak.Dotnet.Sdk/Dtos/Auth/KeycloakAuthConfig.cs:21]: Change the suffix to "Composition"

File: src/Keycloak.Dotnet.Sdk/Dtos/Realms/RealmDto.cs [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+    [JsonPropertyName("config")]
+    public IdentityProviderMappersConfig? Config { get; set; }
+}
+
+public record IdentityProviderMappersConfig
+{
+    [JsonPropertyName("syncMode")]
+    public string? SyncMode { get; set; }
+}
```
Comment (l3oferreira) [inline src/Keycloak.Dotnet.Sdk/Dtos/Realms/RealmDto.cs:136]: Same pattern as in other records. Or subclasses with Composition suffix, or refact other places to use like this.

**Review by l3oferreira — CHANGES_REQUESTED** [review]

File: src/Keycloak.Dotnet.Sdk/Dtos/Realms/RealmDto.cs (hunk 1 above)
Comment (Matrix2100) [reply src/Keycloak.Dotnet.Sdk/Dtos/Realms/RealmDto.cs:45]: feito

File: src/Keycloak.Dotnet.Sdk/Dtos/Auth/KeycloakAuthConfig.cs (hunk 2 above)
Comment (Matrix2100) [reply src/Keycloak.Dotnet.Sdk/Dtos/Auth/KeycloakAuthConfig.cs:21]: Done

File: src/Keycloak.Dotnet.Sdk/Dtos/Realms/RealmDto.cs (hunk 3 above)
Comment (Matrix2100) [reply src/Keycloak.Dotnet.Sdk/Dtos/Realms/RealmDto.cs:136]: Done

**Pull request merged**